src/ImpactEvaluator.sol

```diff
@@ -100,37 +100,24 @@ contract ImpactEvaluator is AccessControl, Balances {
                 roundIndex == previousRoundIndex,
             "Can only score previous round"
         );
-
-        uint sumOfScores = validateScores(scores);
-        reward(addresses, scores);
-        previousRoundTotalScores += sumOfScores;
-    }
-
-    function validateScores(uint64[] memory scores) public view returns (uint) {
-        uint64 sum = 0;
-        for (uint i = 0; i < scores.length; i++) {
-            sum += scores[i];
-        }
-        require(sum <= MAX_SCORE, "Sum of scores too big");
-        require(
-            sum + previousRoundTotalScores <= MAX_SCORE,
-            "Sum of scores including historic too big"
-        );
-        return sum;
-    }
-
-    function reward(
-        address payable[] memory addresses,
-        uint64[] memory scores
-    ) private {
+        
+        uint64 sumOfScores = 0;
         for (uint i = 0; i < addresses.length; i++) {
+            uint64 score = scores[i];
             address payable participant = addresses[i];
-            uint amount = (scores[i] * previousRoundRoundReward) / MAX_SCORE;
+            sumOfScores += score;
+            uint amount = (score * previousRoundRoundReward) / MAX_SCORE;
             if (participant != 0x000000000000000000000000000000000000dEaD) {
                 increaseParticipantBalance(participant, amount);
             }
             previousRoundRemainingReward -= amount;
         }
+        require(sumOfScores <= MAX_SCORE, "Sum of scores too big");
```

Comment:
Isn't this check redundant?

Assuming we use unsigned integers everywhere, if `sumOfScores + previousRoundTotalScores <= MAX_SCORE` then surely `sumOfScores <= MAX_SCORE` must be true as well.